Build warnings since zlib-intel update

https://github.com/dotnet/corefx/pull/32732 introduced the following warnings in master:

```
c:\git\corefx\src\native\windows\clrcompression\zlib-intel\inflate.h(141): warning C4267: 'function': conversion from 'size_t' to 'long', possible loss of data (compiling source file C:\git\corefx\src\Native\Windows\clrcompression\zlib-intel\inffast.c) [C:\git\corefx\artifacts\obj\native\netcoreapp-Windows_NT-Debug-x64-aot\clrcompression\clrcompression.vcxproj] [C:\git\corefx\src\Native\build-native.proj]
c:\git\corefx\src\native\windows\clrcompression\zlib-intel\inflate.h(150): warning C4267: 'function': conversion from 'size_t' to 'uInt', possible loss of data (compiling source file C:\git\corefx\src\Native\Windows\clrcompression\zlib-intel\inffast.c) [C:\git\corefx\artifacts\obj\native\netcoreapp-Windows_NT-Debug-x64-aot\clrcompression\clrcompression.vcxproj] [C:\git\corefx\src\Native\build-native.proj]
c:\git\corefx\src\native\windows\clrcompression\zlib-intel\inflate.h(154): warning C4267: 'function': conversion from 'size_t' to 'uInt', possible loss of data (compiling source file C:\git\corefx\src\Native\Windows\clrcompression\zlib-intel\inffast.c) [C:\git\corefx\artifacts\obj\native\netcoreapp-Windows_NT-Debug-x64-aot\clrcompression\clrcompression.vcxproj] [C:\git\corefx\src\Native\build-native.proj]
c:\git\corefx\src\native\windows\clrcompression\zlib-intel\inflate.h(174): warning C4267: '-=': conversion from 'size_t' to 'uInt', possible loss of data (compiling source file C:\git\corefx\src\Native\Windows\clrcompression\zlib-intel\inffast.c) [C:\git\corefx\artifacts\obj\native\netcoreapp-Windows_NT-Debug-x64-aot\clrcompression\clrcompression.vcxproj] [C:\git\corefx\src\Native\build-native.proj]
c:\git\corefx\src\native\windows\clrcompression\zlib-intel\inflate.h(189): warning C4267: '-=': conversion from 'size_t' to 'unsigned int', possible loss of data (compiling source file C:\git\corefx\src\Native\Windows\clrcompression\zlib-intel\inffast.c) [C:\git\corefx\artifacts\obj\native\netcoreapp-Windows_NT-Debug-x64-aot\clrcompression\clrcompression.vcxproj] [C:\git\corefx\src\Native\build-native.proj]
c:\git\corefx\src\native\windows\clrcompression\zlib-intel\inflate.h(190): warning C4267: '+=': conversion from 'size_t' to 'unsigned int', possible loss of data (compiling source file C:\git\corefx\src\Native\Windows\clrcompression\zlib-intel\inffast.c) [C:\git\corefx\artifacts\obj\native\netcoreapp-Windows_NT-Debug-x64-aot\clrcompression\clrcompression.vcxproj] [C:\git\corefx\src\Native\build-native.proj]
c:\git\corefx\src\native\windows\clrcompression\zlib-intel\inffast.c(102): warning C4244: '=': conversion from '__int64' to 'unsigned int', possible loss of data [C:\git\corefx\artifacts\obj\native\netcoreapp-Windows_NT-Debug-x64-aot\clrcompression\clrcompression.vcxproj] [C:\git\corefx\src\Native\build-native.proj]
c:\git\corefx\src\native\windows\clrcompression\zlib-intel\inffast.c(236): warning C4244: '=': conversion from '__int64' to 'unsigned int', possible loss of data [C:\git\corefx\artifacts\obj\native\netcoreapp-Windows_NT-Debug-x64-aot\clrcompression\clrcompression.vcxproj] [C:\git\corefx\src\Native\build-native.proj]
c:\git\corefx\src\native\windows\clrcompression\zlib-intel\inflate.h(141): warning C4267: 'function': conversion from 'size_t' to 'long', possible loss of data (compiling source file C:\git\corefx\src\Native\Windows\clrcompression\zlib-intel\inflate.c) [C:\git\corefx\artifacts\obj\native\netcoreapp-Windows_NT-Debug-x64-aot\clrcompression\clrcompression.vcxproj] [C:\git\corefx\src\Native\build-native.proj]
c:\git\corefx\src\native\windows\clrcompression\zlib-intel\inflate.h(150): warning C4267: 'function': conversion from 'size_t' to 'uInt', possible loss of data (compiling source file C:\git\corefx\src\Native\Windows\clrcompression\zlib-intel\inflate.c) [C:\git\corefx\artifacts\obj\native\netcoreapp-Windows_NT-Debug-x64-aot\clrcompression\clrcompression.vcxproj] [C:\git\corefx\src\Native\build-native.proj]
c:\git\corefx\src\native\windows\clrcompression\zlib-intel\inflate.h(154): warning C4267: 'function': conversion from 'size_t' to 'uInt', possible loss of data (compiling source file C:\git\corefx\src\Native\Windows\clrcompression\zlib-intel\inflate.c) [C:\git\corefx\artifacts\obj\native\netcoreapp-Windows_NT-Debug-x64-aot\clrcompression\clrcompression.vcxproj] [C:\git\corefx\src\Native\build-native.proj]
c:\git\corefx\src\native\windows\clrcompression\zlib-intel\inflate.h(174): warning C4267: '-=': conversion from 'size_t' to 'uInt', possible loss of data (compiling source file C:\git\corefx\src\Native\Windows\clrcompression\zlib-intel\inflate.c) [C:\git\corefx\artifacts\obj\native\netcoreapp-Windows_NT-Debug-x64-aot\clrcompression\clrcompression.vcxproj] [C:\git\corefx\src\Native\build-native.proj]
c:\git\corefx\src\native\windows\clrcompression\zlib-intel\inflate.h(189): warning C4267: '-=': conversion from 'size_t' to 'unsigned int', possible loss of data (compiling source file C:\git\corefx\src\Native\Windows\clrcompression\zlib-intel\inflate.c) [C:\git\corefx\artifacts\obj\native\netcoreapp-Windows_NT-Debug-x64-aot\clrcompression\clrcompression.vcxproj] [C:\git\corefx\src\Native\build-native.proj]
c:\git\corefx\src\native\windows\clrcompression\zlib-intel\inflate.h(190): warning C4267: '+=': conversion from 'size_t' to 'unsigned int', possible loss of data (compiling source file C:\git\corefx\src\Native\Windows\clrcompression\zlib-intel\inflate.c) [C:\git\corefx\artifacts\obj\native\netcoreapp-Windows_NT-Debug-x64-aot\clrcompression\clrcompression.vcxproj] [C:\git\corefx\src\Native\build-native.proj]
c:\git\corefx\src\native\windows\clrcompression\zlib-intel\inflate.c(719): warning C4244: '=': conversion from '__int64' to 'unsigned int', possible loss of data [C:\git\corefx\artifacts\obj\native\netcoreapp-Windows_NT-Debug-x64-aot\clrcompression\clrcompression.vcxproj] [C:\git\corefx\src\Native\build-native.proj]
c:\git\corefx\src\native\windows\clrcompression\zlib-intel\inflate.c(878): warning C4244: '=': conversion from '__int64' to 'unsigned int', possible loss of data [C:\git\corefx\artifacts\obj\native\netcoreapp-Windows_NT-Debug-x64-aot\clrcompression\clrcompression.vcxproj] [C:\git\corefx\src\Native\build-native.proj]
c:\git\corefx\src\native\windows\clrcompression\zlib-intel\inflate.c(946): warning C4244: '=': conversion from '__int64' to 'unsigned int', possible loss of data [C:\git\corefx\artifacts\obj\native\netcoreapp-Windows_NT-Debug-x64-aot\clrcompression\clrcompression.vcxproj] [C:\git\corefx\src\Native\build-native.proj]
c:\git\corefx\src\native\windows\clrcompression\zlib-intel\inflate.c(949): warning C4244: '=': conversion from '__int64' to 'unsigned long', possible loss of data [C:\git\corefx\artifacts\obj\native\netcoreapp-Windows_NT-Debug-x64-aot\clrcompression\clrcompression.vcxproj] [C:\git\corefx\src\Native\build-native.proj]
c:\git\corefx\src\native\windows\clrcompression\zlib-intel\inflate.c(1090): warning C4244: '=': conversion from '__int64' to 'unsigned int', possible loss of data [C:\git\corefx\artifacts\obj\native\netcoreapp-Windows_NT-Debug-x64-aot\clrcompression\clrcompression.vcxproj] [C:\git\corefx\src\Native\build-native.proj]
c:\git\corefx\src\native\windows\clrcompression\zlib-intel\inflate.c(1195): warning C4244: '=': conversion from '__int64' to 'unsigned int', possible loss of data [C:\git\corefx\artifacts\obj\native\netcoreapp-Windows_NT-Debug-x64-aot\clrcompression\clrcompression.vcxproj] [C:\git\corefx\src\Native\build-native.proj]
c:\git\corefx\src\native\windows\clrcompression\zlib-intel\inflate.c(1201): warning C4244: '=': conversion from '__int64' to 'unsigned long', possible loss of data [C:\git\corefx\artifacts\obj\native\netcoreapp-Windows_NT-Debug-x64-aot\clrcompression\clrcompression.vcxproj] [C:\git\corefx\src\Native\build-native.proj]
c:\git\corefx\src\native\windows\clrcompression\zlib-intel\inflate.c(1238): warning C4244: '=': conversion from '__int64' to 'unsigned int', possible loss of data [C:\git\corefx\artifacts\obj\native\netcoreapp-Windows_NT-Debug-x64-aot\clrcompression\clrcompression.vcxproj] [C:\git\corefx\src\Native\build-native.proj]
c:\git\corefx\src\native\windows\clrcompression\zlib-intel\inflate.c(1247): warning C4244: '=': conversion from '__int64' to 'unsigned int', possible loss of data [C:\git\corefx\artifacts\obj\native\netcoreapp-Windows_NT-Debug-x64-aot\clrcompression\clrcompression.vcxproj] [C:\git\corefx\src\Native\build-native.proj]
c:\git\corefx\src\native\windows\clrcompression\zlib-intel\inflate.c(1312): warning C4244: '=': conversion from '__int64' to 'unsigned int', possible loss of data [C:\git\corefx\artifacts\obj\native\netcoreapp-Windows_NT-Debug-x64-aot\clrcompression\clrcompression.vcxproj] [C:\git\corefx\src\Native\build-native.proj]
c:\git\corefx\src\native\windows\clrcompression\zlib-intel\inflate.c(941): warning C4244: 'initializing': conversion from '__int64' to 'unsigned long', possible loss of data [C:\git\corefx\artifacts\obj\native\netcoreapp-Windows_NT-Debug-x64-aot\clrcompression\clrcompression.vcxproj] [C:\git\corefx\src\Native\build-native.proj]
c:\git\corefx\src\native\windows\clrcompression\zlib-intel\inflate.c(1192): warning C4244: 'initializing': conversion from '__int64' to 'unsigned long', possible loss of data [C:\git\corefx\artifacts\obj\native\netcoreapp-Windows_NT-Debug-x64-aot\clrcompression\clrcompression.vcxproj] [C:\git\corefx\src\Native\build-native.proj]
c:\git\corefx\src\native\windows\clrcompression\zlib-intel\x86.c(49): warning C4047: 'function': 'int *' differs in levels of indirection from 'int (*)[4]' [C:\git\corefx\artifacts\obj\native\netcoreapp-Windows_NT-Debug-x64-aot\clrcompression\clrcompression.vcxproj] [C:\git\corefx\src\Native\build-native.proj]
c:\git\corefx\src\native\windows\clrcompression\zlib-intel\x86.c(49): warning C4024: '__cpuid': different types for formal and actual parameter 1 [C:\git\corefx\artifacts\obj\native\netcoreapp-Windows_NT-Debug-x64-aot\clrcompression\clrcompression.vcxproj] [C:\git\corefx\src\Native\build-native.proj]
```

cc @stephentoub @vkvenkat

--- a/src/Native/Windows/clrcompression/zlib-intel/inffast.c
+++ b/src/Native/Windows/clrcompression/zlib-intel/inffast.c
@@ -99,7 +99,7 @@ unsigned start;         /* inflate()'s starting value for strm->avail_out */
         }
         
         if (out >= end) {
-            state->wnext = out - (state->window + state->wsize);
+            state->wnext = (unsigned)(out - (state->window + state->wsize));
             window_output_flush(strm);
             out = state->window + state->wsize + state->wnext;
             if (strm->avail_out == 0)
@@ -233,7 +233,7 @@ unsigned start;         /* inflate()'s starting value for strm->avail_out */
     strm->next_in = in;
     strm->avail_in = (unsigned)(in < last ? 5 + (last - in) : 5 - (in - last));
 
-    state->wnext = out - (state->window + state->wsize);
+    state->wnext = (unsigned)(out - (state->window + state->wsize));
 
     state->hold = hold;
     state->bits = bits;

--- a/src/Native/Windows/clrcompression/zlib-intel/inflate.c
+++ b/src/Native/Windows/clrcompression/zlib-intel/inflate.c
@@ -505,7 +505,7 @@ unsigned copy;
 /* Restore state from registers in inflate() */
 #define RESTORE() \
     do { \
-        state->wnext = put - (state->window + state->wsize);\
+        state->wnext = (unsigned)(put - (state->window + state->wsize));\
         strm->avail_out = left; \
         strm->next_in = next; \
         strm->avail_in = have; \
@@ -938,7 +938,7 @@ int flush;
             copy = state->length;
             if (copy) {
                 unsigned char *end = state->window + (state->wsize * 4);
-                unsigned long diff = end - put;
+                unsigned diff = (unsigned)(end - put);
 
                 if (copy > have) copy = have;
                 if (copy > diff) {
@@ -946,7 +946,7 @@ int flush;
                         RESTORE();
                         window_output_flush(strm);
                         LOAD();
-                        diff = end - put;
+                        diff = (unsigned)(end - put);
                     }
                     if (copy > diff) copy = diff;
                 }
@@ -1189,7 +1189,7 @@ int flush;
             state->mode = MATCH;
             case MATCH: {
                 unsigned char *end = state->window + (state->wsize * 4);
-                unsigned long buf_left = end - put;
+                unsigned buf_left = (unsigned)(end - put);
                 copy = state->length;
 
 		RESTORE();
@@ -1198,7 +1198,7 @@ int flush;
                         /* relies on RESTORE() above with no changes to those vars */
                         window_output_flush(strm);
                         LOAD();
-                        buf_left = end - put;
+                        buf_left = (unsigned)(end - put);
                     }
 		    if (copy > buf_left) copy = buf_left;
                 }

--- a/src/Native/Windows/clrcompression/zlib-intel/inflate.h
+++ b/src/Native/Windows/clrcompression/zlib-intel/inflate.h
@@ -132,7 +132,7 @@ struct inflate_state {
 
 
 static inline void inf_crc_copy(z_streamp strm, unsigned char FAR *const dst,
-        const unsigned char FAR *const src, size_t len)
+        const unsigned char FAR *const src, uInt len)
 {
     struct inflate_state *const state = (struct inflate_state *const)strm->state;
 
@@ -159,7 +159,7 @@ static inline void window_output_flush(z_streamp strm)
     struct inflate_state *const state = (struct inflate_state *const)strm->state;
 
     size_t woff, roff, copysz;
-    size_t nexto_len;
+    uInt nexto_len;
 
     if (state->wnext > strm->avail_out) {
         nexto_len = strm->avail_out;

--- a/src/Native/Windows/clrcompression/zlib-intel/x86.c
+++ b/src/Native/Windows/clrcompression/zlib-intel/x86.c
@@ -46,7 +46,7 @@ void x86_check_features(void)
     once = 1;
 
 #ifdef _MSC_VER
-    __cpuid(&regs, 1);
+    __cpuid(regs, 1);
 #else
     __cpuid(1, regs[A], regs[B], regs[C], regs[D]);
 #endif

